src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -951,30 +955,34 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			}
 		}
 
-		const expiry = profile.options.expiresOn;
-		if (typeof expiry === 'number' && !Number.isNaN(expiry)) {
-			const currentTime = new Date().getTime() / 1000;
-			const maxTolerance = 2 * 60; // two minutes
-			if (expiry - currentTime < maxTolerance) {
-				this._logService.info(`Access token expired for connection ${profile.id} with uri ${uri}`);
-				try {
-					const connectionResultPromise = this.connect(profile, uri);
-					this._uriToReconnectPromiseMap[uri] = connectionResultPromise;
-					const connectionResult = await connectionResultPromise;
-					if (!connectionResult) {
-						this._logService.error(`Failed to refresh connection ${profile.id} with uri ${uri}, invalid connection result.`);
-						throw new Error(nls.localize('connection.invalidConnectionResult', "Connection result is invalid"));
-					} else if (!connectionResult.connected) {
-						this._logService.error(`Failed to refresh connection ${profile.id} with uri ${uri}, error code: ${connectionResult.errorCode}, error message: ${connectionResult.errorMessage}`);
-						throw new Error(nls.localize('connection.refreshAzureTokenFailure', "Failed to refresh Azure account token for connection"));
+		if (connectionProfile.authenticationType === Constants.AuthenticationType.AzureMFA) {
+			const expiry = connectionProfile.options.expiresOn;
+			if (typeof expiry === 'number' && !Number.isNaN(expiry)) {
+				const currentTime = new Date().getTime() / 1000;
+				const maxTolerance = 2 * 60; // two minutes
+				if (expiry - currentTime < maxTolerance) {
+					this._logService.info(`Access token expired for connection ${connectionProfile.id} with uri ${uri}`);
+					try {
+						const connectionResultPromise = this.connect(connectionProfile, uri);
+						this._uriToReconnectPromiseMap[uri] = connectionResultPromise;
+						const connectionResult = await connectionResultPromise;
+						if (!connectionResult) {
```

Comment:
A good follow up task would be to fix the return type for these methods...although TBH I don't know why returning undefined from these is a valid scenario in the first place. But that'd likely be a larger task to fix so at least making the return types accurate would be a huge help for stuff like this.

Reply:
As I see this function may return false, it's not bound to return true. So, the only way would be to throw an error when we 'expect' to be able to refresh access token. Let me know if you had some other thought!

Reply:
I meant more for the calls you're making here - i.e. this.connect. It's currently returning undefined which seems odd - if it can't connect then it seems better to throw since that's pretty standard behavior.

Reply:
It returns undefined for a canceled firewall dialog; I wouldn't change anything just yet.